toolchain/mfc/test/test.py

```diff
@@ -241,6 +248,10 @@ def handle_case(case: TestCase, devices: typing.Set[int]):
     global errors
 
     nAttempts = 0
+    if ARG('single'):
+        max_attempts = max(ARG('max_attempts'), 20)
```

Comment:
where does this 20 come from? this seems like way too many attempts

Reply:
i think you have a problem on this line below:

`            if nAttempts < ARG("max_attempts"):`

which, since you are prescribing your own arguments for `max_attempts` should really replace as 

`            if nAttempts < max_attempts:`

though see my comment above, it seems like altering max attempts for single precision doesn't make much sense (and 20 certainly shouldn't be necessary).

Reply:
The reason for this was because I noticed that sometimes post process will have NaNs. If you run the test 1 or more times, then all of a sudden the NaNs will go away. Since it was behaving in this non deterministic way, I assumed that some race condition was being introduced somewhere when running the test suite or test all in single precision so the best idea would be to just set max retries to a high number. This only happens in ./mfc.sh test -a n anyway and I dont think it's an actual issue with the simulation(since I visualized things) and everything looks good. Is this okay?

Reply:
is `--single` in the test suite? I know we discussed benchmarking before, which will be next, but shouldn't we run the code through the single precision test suite? (at least for one or two runners)

Reply:
So add a single precision test to CI?

Reply:
yes ./mfc.sh test --single is a thing